src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1461,6 +1536,135 @@ SliceLastMatmulMultiply::SliceLastMatmulMultiply() {
     register_matcher(std::make_shared<opp::Matcher>(res, "SliceLastMatmulMultiply"), std::move(callback));
 }
 
+ConvToMatmul::ConvToMatmul(Context::Ref ctx) {
+    auto param = opp::wrap_type<ov::op::v0::Parameter>();
+    auto convert = opp::optional<ov::op::v0::Convert>({param->output(0)});
+    auto param2 = opp::wrap_type<ov::op::v0::Parameter>();
+    auto multiply = opp::wrap_type<ov::op::v1::Multiply>({convert, param2});
+    auto tr_input = opp::any_input();
+    auto transpose_in = opp::wrap_type<ov::op::v1::Transpose>({tr_input, opp::any_input()});
+    auto conv = opp::wrap_type<ov::op::v1::Convolution>({transpose_in, multiply});
+    auto transpose_out = opp::wrap_type<ov::op::v1::Transpose>({conv, opp::any_input()});
+
+    // Note: Use [=] to make sure the above objects stay alive in the callback
+    auto callback = [=](ov::pass::pattern::Matcher& m) {
+        auto& node_to_output = m.get_pattern_value_map();
+
+        auto matched_node_param = node_to_output.at(param).get_node_shared_ptr();
+        auto matched_node_param2 = node_to_output.at(param2).get_node_shared_ptr();
+        auto matched_node_convert = node_to_output.at(convert).get_node_shared_ptr();
+        auto matched_node_tr_input = node_to_output.at(tr_input).get_node_shared_ptr();
+        auto matched_node_transpose_in = node_to_output.at(transpose_in).get_node_shared_ptr();
+        auto matched_node_transpose_out = node_to_output.at(transpose_out).get_node_shared_ptr();
+        auto matched_node_multiply = node_to_output.at(multiply).get_node_shared_ptr();
+        auto matched_node_conv = node_to_output.at(conv).get_node_shared_ptr();
+
+        auto matched_param = std::static_pointer_cast<ov::op::v0::Parameter>(matched_node_param);
+        auto matched_param2 = std::static_pointer_cast<ov::op::v0::Parameter>(matched_node_param2);
+
+        const auto& shape = matched_param->get_shape();
+        const auto& shape2 = matched_param2->get_shape();
+        const auto& tr_in_shape = matched_node_transpose_in->input(0).get_shape();
+        const auto& tr_out_shape = matched_node_transpose_out->output(0).get_shape();
+
+        if (matched_param->get_element_type() == ov::element::i4 &&
+            matched_param2->get_element_type() == ov::element::f32 && shape.size() == 4 && shape2.size() == 4 &&
+            shape[2] == 1 && shape[3] == 1 && shape2[2] == 1 && shape2[3] == 1 && tr_in_shape.size() == 4 &&
+            tr_out_shape.size() == 4 && tr_in_shape[0] == 1 && tr_in_shape[1] == 1 && tr_out_shape[0] == 1 &&
+            tr_out_shape[1] == 1) {
```

Comment:
let's add simple mnemonic functions to check these shapes, otherwise it is hard to follow.

What I mean - you check multiple shape vectors here, `shape`, `shape2`,  `tr_in_shape`, `tr_out_shape`. Please write individual checkers for those and tiny lambdas and then call them from your `if`.

Reply:
Done